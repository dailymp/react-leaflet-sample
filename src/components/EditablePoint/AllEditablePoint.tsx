import * as React from "react";
import { Point } from "../../common/Point";
import { EditablePointList } from "./EditablePointList";

interface Props {
    pins: Point[];
    onEditPins: (newPins: Point[]) => void;
}

export const AllEditablePoints = (props: Props) => {
    const [showPointList, setShowPointList] = React.useState(false);

    return(
        <>
        <h1>Edit points</h1>
        {
            showPointList ?
                <>
                    <EditablePointList {...props} setShowPointList={setShowPointList} />
                </>
                : <button onClick={evt => setShowPointList(true)}>Edit Point List</button>
        }
        </>
    );
}
