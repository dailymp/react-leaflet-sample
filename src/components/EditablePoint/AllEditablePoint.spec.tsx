import * as React from 'react';
import { shallow } from 'enzyme';
import { AllEditablePoints } from './AllEditablePoint';

describe('Tests for AllEditablePoints component', () => {
  it('should render as expected when passing required properties', () => {
    const props = {
        pins: [
            {x: 1, y:3, hexadecimalColor: '#00e700'},
            {x: 4, y:6, hexadecimalColor: '#00e700'},
        ],
        onEditPins: jest.fn(),
    };

    const component = shallow(<AllEditablePoints
        pins={props.pins}
        onEditPins={props.onEditPins}
    />);

    expect(component).toMatchSnapshot();
  });
});