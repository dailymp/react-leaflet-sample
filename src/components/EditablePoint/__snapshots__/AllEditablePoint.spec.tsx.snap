// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for AllEditablePoints component should render as expected when passing required properties 1`] = `
<Fragment>
  <h1>
    Edit points
  </h1>
  <button
    onClick={[Function]}
  >
    Edit Point List
  </button>
</Fragment>
`;
